igvc_platform/src/joystick_driver/joystick_driver.h

```diff
@@ -0,0 +1,132 @@
+#ifndef PROJECT_JOYSTICK_DRIVER_H
+#define PROJECT_JOYSTICK_DRIVER_H
+
+#include <igvc_msgs/velocity_pair.h>
+#include <ros/ros.h>
+#include <sensor_msgs/Imu.h>
+#include <sensor_msgs/Joy.h>
+
+enum ControlStyle
+{
+  direction_velocity_control,
+  smooth_control,
+  tank_control
+};
+
+struct SmoothControlConfig
+{
+  double k1;
+  double k2;
+  double axle_length;
+  double velocity;
+  double w;
+  double max_w;
+  double delta;
+  double camera;
+  double camera_move_rate;
+  bool camera_initialized;
+};
+
+struct TankControlConfig
+{
+  double max_velocity;
+  double velocity_increment;
+  double velocity;
+};
+
+struct AccelerationLimits
+{
+  double wheel_max_accel;
+  double tangent_max_accel;
+  bool enable;
+};
+
+struct DirectionVelocityConfig
+{
+  double pivot_limit;
+  double max_velocity;
+  double velocity_increment;
+  double velocity;
+};
+
+struct Axes
+{
+  double left_x;
+  double left_y;
+  double right_x;
+  double right_y;
+};
+
+struct JoyMap
+{
+  int a = 0;
```

Comment:
a_button, applies to all the ones below this

Reply:
Fixed